### PR TITLE
Only clean cache directories that we create

### PR DESCRIPTION
**Pull request by WestleyK**
## Description of the Pull Request (PR):

This PR fixes the issue when you set `SINGULARITY_CACHEDIR` to (for example your home directory `$HOME`), then run `singularity cache clean --all`, not it will not remove your home directory. Instead this will only remove the directories that we create, eg. `library`, `shub`, `oci`, `oci-tmp`.

This is most likely a temporary fix, until https://github.com/sylabs/singularity/pull/3623 gets merged.

## This fixes or addresses the following GitHub issues:

- Fixes #

<br>

### CODE REVIEW

**Comment by gvallee** [comment]
If I remember correctly, this will cause a failure in `cmd/internal/cli/actions_linux.go` when running `make testall` on Ubuntu (to be confirmed). The following link may help: https://github.com/sylabs/singularity/pull/3595/files#diff-896d1f4723d0a072f845c405328b7f1e

**Review by ikaneshiro — APPROVED** [review]
lgtm will let @gvallee have a look too before merging

File: internal/pkg/client/cache/dir.go [hunk 1]
```diff
@@ -42,12 +42,17 @@ func Root() string {
...
 func Clean() error {
-	sylog.Debugf("Removing: %v", Root())
-
-	if err := os.RemoveAll(Root()); err != nil {
-		return fmt.Errorf("unable to clean all cache: %s", err)
+	// TODO: add oras here
+	cacheDirs := append([]string{}, Library(), OciTemp(), OciBlob(), Shub())
+
+	for _, c := range cacheDirs {
+		sylog.Debugf("Removeing: %s ...", c)
```
Comment (gvallee) [inline internal/pkg/client/cache/dir.go]:
```suggestion
		sylog.Debugf("Removing: %s ...", c)
```

**Review by gvallee — CHANGES_REQUESTED** [review]
Exactly what I was suggesting, thanks for doing it. I only found a minor typo.

**Review by gvallee — APPROVED** [review]

**Pull request merged**